src/tools.ts

```diff
@@ -226,3 +226,45 @@ export function displayWebGLContextLostWarning(container: HTMLElement | string)
   actualContainer.appendChild(errorMessageDiv);
   // throw new Error(webglError);
 }
+
+/**
+ * Return true if the provided piece of style expression has the form ["get", "name:XX"]
+ * with XX benig a 2-letter is code for languages
+ */
+function isGetNameLanguage(subExpr: unknown): boolean {
+  if (!Array.isArray(subExpr)) return false;
+  if (subExpr.length !== 2) return false;
+  if (subExpr[0] !== "get") return false;
+  if (typeof subExpr[1] !== "string") return false;
+  if (!subExpr[1].startsWith("name:")) return false;
+
+  return true;
+}
+
+/**
+ * In a text-field style property (as an object, not a string) the langages that are specified as
+ * ["get", "name:XX"] are replaced by the proved replacer (also an object).
+ * This replacement happened regardless of how deep in the object the flag is.
+ * Note that it does not replace the occurences of ["get", "name"] (local names)
```

Comment:
Is this correct? Why wouldn't we replace the occurrences of local names?

Reply:
This is because in the official style, the labels are `name:en`. The local `name` is sometimes used, but always as a fallback in case `name:en` is not available. In this approach we only replace `name:xx` but keep the `name` as fallback.